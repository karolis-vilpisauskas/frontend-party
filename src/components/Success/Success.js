import React from "react";
import PropTypes from "prop-types";
import "./Success.scss";

const Success = ({ message }) => {
  return (
    <div
      data-testid="success"
      className={message ? "success is-active" : "success"}
    >
      {message}
    </div>
  );
};

export default Success;

Success.propTypes = {
  message: PropTypes.string
};
