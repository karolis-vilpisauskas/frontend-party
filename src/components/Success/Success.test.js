import '@testing-library/jest-dom/extend-expect'
import React from "react";
import { cleanup, render } from "@testing-library/react";
import Success from "./Success";
afterEach(cleanup);

const createTestProps = props => ({
  ...props
});

const renderTest = () => {
  const props = createTestProps();
  const { getByTestId } = render(
    <Success {...props}>
    </Success>
  );
  const container = getByTestId("success");
  return {
    getByTestId,
    container
  };
};

describe("Success", () => {
  describe("rendering", () => {
    test("it renders without crashing", () => {
      const { container } = renderTest();
      expect(container).toMatchSnapshot();
    });
  });
});
