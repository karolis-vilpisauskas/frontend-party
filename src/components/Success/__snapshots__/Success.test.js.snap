// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success rendering it renders without crashing 1`] = `
<div
  class="success"
  data-testid="success"
/>
`;
